benchmark.js

```diff
@@ -123,6 +123,44 @@
 
   /*--------------------------------------------------------------------------*/
 
+
+  var isAsyncFunction;
+  try {
+    var asyncFunctionPrototype = eval('Object.getPrototypeOf(async function () { })');
```

Comment:
What is `fn` itself just returns `Promise`?

```js
suite.add('sh-syntax', () => print(text, shOptions)) // `print` returns `Promise`
```

Reply:
I really thought about it alot. And it is formally a sync Function. If I add code which detects if it is returning a Promise or not, then it would could make the benchmark measuring wrong. If you know it is a Promise but not make it a formal async function, then you should wrap it yourself in the standard benchmark.js defer structure.